indico/modules/events/registration/controllers/management/regforms.py

```diff
@@ -252,6 +255,37 @@ def _process(self):
                                                     regform=self.regform)
 
 
+class RHRegistrationFormNotificationPreview(RHManageRegFormBase):
+    """Preview registration emails."""
+
+    @no_autoflush
+    @use_kwargs({
+        'message': fields.String(load_default=''),
+        'state': fields.String(load_default=RegistrationState.pending.name)
+    })
+    def _process(self, message, state):
+        self.commit = False
+        state = RegistrationState.get(state)
+        if state == RegistrationState.pending:
+            self.regform.message_pending = message
+        elif state == RegistrationState.unpaid:
+            self.regform.message_unpaid = message
+        elif state == RegistrationState.complete:
+            self.regform.message_complete = message
```

Comment:
This would be a nice place to use some modern Python ;)

```py
match state:
    case RegistrationState.pending:
         self.regform.message_pending = message
    case RegistrationState.unpaid:
         self.regform.message_unpaid = message
    case RegistrationState.complete:
         self.regform.message_complete = message
```

Reply:
WTF! 👀 

HELL YES! AT LAST!